src/modules/SignUp/components/SignUp/components/FormSignUp/FormSignUp.jsx

```diff
@@ -25,59 +23,169 @@ export default class FormSignUp extends Component {
+                    }));
+
+                    Object.keys(this.fieldRefs).forEach((ref) => {
+                        this.fieldRefs[ref].current.forceUpdate();
+                    });
+
+                    this.setState(() => ({
+                        errorServer: [],
+                    }));
+                }
             });
+    }
+
+    usernameValidor(value) {
+        const { errorServer } = this.state;
+        let usernameValid = true;
+        let errorMsg = "";
+
+        if (!value) {
+            usernameValid = false;
+            errorMsg = signUpErrorMsgs.emptyUsername;
+        }
+
+        if (errorServer.includes("21")) {
+            usernameValid = false;
+            errorMsg = signUpErrorMsgs.existUsername;
+        }
+
+        this.setState((prevState) => ({
+            buttonStates: {
+                ...prevState.buttonStates,
+                username: usernameValid,
+            },
+            errorFields: {
+                ...prevState.errorFields,
+                username: errorMsg,
+            },
+        }));
+
+        return usernameValid;
+    }
+
+    emailValidor(value) {
+        const { errorServer } = this.state;
+        const isValid = /.+@.+\..+/i;
+        let emailValid = true;
+        let errorMsg = "";
+
+        if (!isValid.test(value)) {
+            emailValid = false;
+            errorMsg = signUpErrorMsgs.incorrectEmail;
+        }
+
+        if (!value) {
+            emailValid = false;
+            errorMsg = signUpErrorMsgs.emptyEmail;
         }
+
+        if (errorServer.includes("22")) {
+            emailValid = false;
+            errorMsg = signUpErrorMsgs.existEmail;
+        }
+
+        this.setState((prevState) => ({
+            buttonStates: {
+                ...prevState.buttonStates,
+                email: emailValid,
+            },
+            errorFields: {
+                ...prevState.errorFields,
+                email: errorMsg,
+            },
+        }));
+
+        return emailValid;
+    }
+
+    passwordValidor() {
+        const { errorServer, formData } = this.state;
+        let passwordValid = true;
+        let errorMsg = "";
+
+        if (errorServer.includes("12")) {
+            passwordValid = false;
+            errorMsg = signUpErrorMsgs.emptyPassword;
+        }
+
+        if (formData.password !== formData.retypePassword) {
+            passwordValid = false;
+            errorMsg = signUpErrorMsgs.mismatchPassword;
+        }
+
+        this.setState((prevState) => ({
+            buttonStates: {
+                ...prevState.buttonStates,
+                password: passwordValid,
+            },
+            errorFields: {
+                ...prevState.errorFields,
+                password: errorMsg,
+            },
+        }));
+
+        return passwordValid;
+    }
+
+    checkButtonStates() {
+        const { buttonStates } = this.state;
+        if (Object.values(buttonStates).every((elem) => elem === true)) {
```

Comment:
You can replace a conditional expression with `return !Object.values(buttonStates).every((elem) => elem === true);`.